harness/tests/experiment/pytorch/test_pytorch_trial.py

```diff
@@ -414,19 +421,41 @@ def make_workloads2() -> workload.Stream:
             "legacy_on_training_epochs_start_calls": 1
         }
 
-    def test_context(self) -> None:
+    @pytest.mark.parametrize(
+        "lr_scheduler_step_mode", [mode.value for mode in pytorch.LRScheduler.StepMode]
+    )
+    def test_context(
+        self,
+        lr_scheduler_step_mode,
+    ) -> None:
         def make_workloads() -> workload.Stream:
             trainer = utils.TrainAndValidate()
             yield from trainer.send(steps=1, validation_freq=1, scheduling_unit=1)
 
+        hparams = self.hparams.copy()
+        hparams["lr_scheduler_step_mode"] = lr_scheduler_step_mode
+
         controller = utils.make_trial_controller_from_trial_implementation(
             trial_class=pytorch_xor_model.XORTrialAccessContext,
-            hparams=self.hparams,
+            hparams=hparams,
             workloads=make_workloads(),
             trial_seed=self.trial_seed,
         )
         controller.run()
 
+    @pytest.mark.skip("WIP")  # FIXME
```

Comment:
i'd say this test probably isn't worth investing time into, since i imagine we'd need a whole suite of unit tests to test the behavior of the workload sequencer/generator. we plan on deprecating the workload sequencer in the near future anyway, so e2e tests are probably enough for this part of the codebase.

Reply:
agreed, the entire `_workload_sequencer.py` file is pure legacy, please don't waste any time testing it.

Reply:
Ok, I'll scrap it.

Reply:
One second thought, I think I'll keep the code but mark it as expected to fail. It can be blown away later with the other legacy code it touches.

Reply:
Please do not do this.  Adding code whose only purpose is to be deleted later just makes the codebase harder to read and harder to understand.

Reply:
Ok, I will remove this code now, but I did not write this code in the first place with the purpose to be deleted later: I wrote it with purpose for my assignment to "write tests; improve coverage".
I made an educated suggestion to mark the code because it is easer to remove code than write it.

Reply:
Well sure, nobody writes code with the purpose of deleting.  I am sorry if I implied that you intended it to be that way; I did not think that was the case.

I have written plenty of code like that fits the category of "intended to be useful but turned out to not be useful right now".  And when that happens it's best to leave it out of the code.

Personally, I have a few branches laying around that I don't want to delete but which I don't want to land either, like this [alternate build system](https://github.com/rb-determined-ai/determined/tree/mkninja) and this [buildcache integration](https://github.com/rb-determined-ai/determined/tree/buildcache).  Both efforts were scrapped because I realized we were actively hiring infra people and they would probably rather write/maintain something entirely of their own creation.

So if you have code that's not useful right now, I'd recommend keeping it around until you know it won't be useful.  I just mean that it should not go into the codebase.